### PR TITLE
Log a shortened version of the enterprise token

### LINKED ISSUE #18249: The token is logged in plain text
**Describe the bug**
This line - https://github.com/metabase/metabase/blob/master/src/metabase/public_settings/premium_features.clj#L66 - logs the token in plain text.

**Logs**
```
{"timeMillis":1633448622856,"thread":"clojure-agent-send-off-pool-2","level":"INFO","loggerName":"metabase.public-settings.metastore","message":"Using this URL to check token: https://store.metabase.com/api/**redacted**/v2/status","endOfBatch":false,"loggerFqcn":"org.apache.logging.slf4j.Log4jLogger","threadId":54,"threadPriority":5}
```

**To Reproduce**
1. Run Metabase with an enterprise token 
2. Browse the logs and look for `Using this URL to check token`

**Expected behavior**
No token should be leaked to the logs.

**Screenshots**
N/A

**Information about your Metabase Installation:**
v1.39.5

**Severity**
High

**Additional context**
N/A

### PR DESCRIPTION
**Pull request by escherize**
fixes #18249 
- when checking that an enterprise token is valid
- test captures the logs, and checks for no token
- in the test, regrettably we print the logs we need to capture

### CODE REVIEW

File: test/metabase/public_settings/premium_features_test.clj [hunk 1]
```diff
@@ -40,8 +41,17 @@
                 :trial    false}))
 
 (deftest fetch-token-status-test
-  (tt/with-temp User [user {:email "admin@example.com"}]
-    (let [token "fa3ebfa3ebfa3ebfa3ebfa3ebfa3ebfa3ebfa3ebfa3ebfa3ebfa3ebfa3ebfa3e"]
+  (tt/with-temp User [_user {:email "admin@example.com"}]
+    (let [token       "fa3ebfa3ebfa3ebfa3ebfa3ebfa3ebfa3ebfa3ebfa3ebfa3ebfa3ebfa3ebfa3e"
+          print-token "fa3e...fa3e"]
```
Comment (noahmoss) [inline test/metabase/public_settings/premium_features_test.clj:46]: nit: maybe use a test token that has different first four & last four characters, just for the extra validation that truncation is happening correctly?

File: test/metabase/public_settings/premium_features_test.clj [hunk 2]
```diff
@@ -4,6 +4,7 @@
             [clojure.test :refer :all]
             [metabase.models.user :refer [User]]
             [metabase.public-settings.premium-features :as premium-features]
+            [metabase.test.util :as tu] ;; Circular reference with mt in here, so use tu
```
Comment (noahmoss) [inline test/metabase/public_settings/premium_features_test.clj:7]: You got a circular reference when using `[metabase.test :as mt]`? I just tried it locally and didn't see that. Curious how that was happening.

**Review by noahmoss — APPROVED** [review]
LGTM, though confused about the circular reference comment (might not be necessary?)

File: test/metabase/public_settings/premium_features_test.clj (hunk 2 above)
Comment (dpsutton) [reply test/metabase/public_settings/premium_features_test.clj:7]: Did you run the tests from the command line or the repl?

Comment (noahmoss) [reply test/metabase/public_settings/premium_features_test.clj:7]: From the command line

Comment (escherize) [reply test/metabase/public_settings/premium_features_test.clj:7]: I just verified, there doesn't seem to be a circular dep issue. Not sure what I was seeing yesterday. I'll fixup now. Thanks @noahmoss

Comment (escherize) [reply test/metabase/public_settings/premium_features_test.clj]: That's right! lsp was just a bit too helpful.

**Pull request merged**